app/models/custom_attribute.rb

```diff
@@ -2,7 +2,17 @@ class CustomAttribute < ApplicationRecord
   belongs_to :resource, :polymorphic => true
   serialize :serialized_value
 
+  def value=(value)
+    self.serialized_value = value
+    self[:value] = value
+  end
+
   def stored_on_provider?
     source == "VC"
   end
+
+  def field_type
+    field_type = serialized_value.class
+    field_type == NilClass ? :string : field_type.to_s.downcase.to_sym
```

Comment:
@alongoldboim if that's the case, it looks like your algorithm won't properly handle underscoring. Can you add tests for all the supported cases?